src/codegen/extensions/swe_bench/swe_bench_wrapper.py

```
@@ -0,0 +1,80 @@
+import shutil
+from collections.abc import Generator
+from typing import Any
+
+from datasets import load_dataset
+
+from codegen.extensions.swe_bench.utils import NO_ENV_SETUP, SWEBenchEntry, SWEBenchEnvSetup, SWEBenchSplit, construct_codebase
+from codegen.sdk.core.codebase import Codebase
+
+
+class SWEBenchWrapper:
+    def __init__(self, remove_after_run: bool = False):
+        print("Loading SWE-bench dataset...")
```

Consider adding proper error handling for dataset loading and repo operations. The current implementation might fail silently in case of network issues or dataset loading problems. Consider wrapping critical operations in try-except blocks and providing meaningful error messages.